middleware/kubernetes/handler.go

```diff
@@ -40,10 +38,8 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 		// Set the zone to this specific request.
 		zone = state.Name()
 	}
-
 	records, extra, _, err := k.routeRequest(zone, state)
-
-	if k.AutoPath.Enabled && k.IsNameError(err) {
+	if k.AutoPath.Enabled && k.IsNameError(err) && (state.QType() == dns.TypeA || state.QType() == dns.TypeAAAA) {
 		p := k.findPodWithIP(state.IP())
 		for p != nil {
```

Comment:
if you mean to replace the `for p != nil {`  I don't think it would work. `continue` I thought only works inside a loop.